core/src/main/scala/scala/pickling/tags/FastTypeTag.scala

```diff
@@ -0,0 +1,269 @@
+package scala.pickling
+package tags
+
+import scala.language.experimental.macros
+import scala.reflect.runtime.{universe => ru}
+
+/**
+ * A "tag" denoting information about a runtime type.
+ * This tag is meant to be extermely efficient for simple runtime checks, avoiding a full reflection overhead, while
+ * also *allowing* full runtime checks.
+ *
+ * Notes:
+ *
+ * 1. Currently the equals method does "stringy" comparison of types.  For a full "conforms" comparison, you'll need to
+ *    fully reify the Type (tpe).
+ * 2. It is vitally important to this class that an tag created using `apply` matches those created using `makeRaw`.
+ *    As such, a few oddities have arisen:
+ *    - Existentials are encoded as `scala.Any` type parameters, rather than erased.  This has many varied
+ *      implications throughout the codebase, and marks a compromise (currently) to keep things working.
+ *    - This class does NOT support refinement types, e.g. `Foo with Baz`.  Indeed, this is not a physical class
```

Comment:
What's the difference?

`Foo with Baz` -> compound?
`Foo with { def x: String }` => refinement?

Why is it called "RefinedType" in the compiler?